operations/helm/charts/mimir-distributed/templates/query-frontend/query-frontend-so.yaml

```diff
@@ -0,0 +1,43 @@
+{{- if .Values.query_frontend.kedaAutoscaling.enabled }}
+apiVersion: keda.sh/v1alpha1
+kind: ScaledObject
+metadata:
+  name: {{ include "mimir.resourceName" (dict "ctx" . "component" "query-frontend") }}
+  labels:
+    {{- include "mimir.labels" (dict "ctx" . "component" "query-frontend") | nindent 4 }}
+  annotations:
+    {{- toYaml .Values.query_frontend.annotations | nindent 4 }}
+  namespace: {{ .Release.Namespace | quote }}
+spec:
+  advanced:
+    horizontalPodAutoscalerConfig:
+      {{- with .Values.query_frontend.kedaAutoscaling.behavior }}
+      behavior:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+  maxReplicaCount: {{ .Values.query_frontend.kedaAutoscaling.maxReplicaCount }}
+  minReplicaCount: {{ .Values.query_frontend.kedaAutoscaling.minReplicaCount }}
+  pollingInterval: {{ .Values.query_frontend.kedaAutoscaling.pollingInterval }}
+  scaleTargetRef:
+    name: {{ include "mimir.resourceName" (dict "ctx" . "component" "query-frontend") }}
+    apiVersion: apps/v1
+    kind: Deployment
+  triggers:
+  - metadata:
+      metricName: query_frontend_cpu_hpa_default
```

Comment:
> Jsonnet still supplies the metricName field. In jsonnet the _default suffix is actually the namespace. I wonder if we should spare ourselves the trouble and just remove it now from the helm templates.

I'd be for removing it since this is a new feature to helm.

> @jhesketh what's your take on how many keda versions we should support? KEDA 2.10 is from March 2023, 2.12 is from September 2023. How painful are KEDA upgrades? Can we require a fairly up-to-date version of KEDA (11 months old; is that even up to date)?

I don't have a particularly strong opinion here, but since this is a new (to helm) feature, I would tend towards just supporting the latest version of KEDA. The jsonnet may work with older versions, but we should prefer the newest.

I don't believe upgrading KEDA is painful, but I have not done so.

Reply:
ok, then let's drop the metricName from all the ScaledObjects. @beatkind can you do that?

Reply:
I agree here. Only supporting the newest version of KEDA makes sense. I will drop the `metricName`field.